iceoryx_hoofs/include/iceoryx_hoofs/posix_wrapper/thread.hpp

```diff
@@ -28,8 +33,64 @@ constexpr uint64_t MAX_THREAD_NAME_LENGTH = 15U;
 
 using ThreadName_t = cxx::string<MAX_THREAD_NAME_LENGTH>;
 
-void setThreadName(pthread_t thread, const ThreadName_t& name) noexcept;
-ThreadName_t getThreadName(pthread_t thread) noexcept;
+/// @todo remove free functions
+void setThreadName(iox_pthread_t thread, const ThreadName_t& name) noexcept;
+ThreadName_t getThreadName(iox_pthread_t thread) noexcept;
+
+enum class ThreadError
+{
+    EMPTY_CALLABLE,
+    INSUFFICIENT_MEMORY,
+    INSUFFICIENT_PERMISSIONS,
+    INSUFFICIENT_RESOURCES,
+    INVALID_ATTRIBUTES,
+    UNDEFINED
+};
+
+class Thread
+{
+  public:
+    using callable_t = cxx::function<void()>;
+
+    Thread(const Thread&) = delete;
+    Thread& operator=(const Thread&) = delete;
+    Thread(Thread&&) = delete;
+    Thread& operator=(Thread&&) = delete;
+
+    ~Thread() noexcept;
+
+    // @todo set name during creation and remove method
+    void setName(const ThreadName_t& name) noexcept;
+
+    /// @brief Returns the name of the thread
+    /// @return An iox::cxx::string containing the name of the thread
+    ThreadName_t getName() noexcept;
```

Comment:
This should be a const method. And when the name can only be set at creation time we can also return here a const reference, so
```cxx
const ThreadName_t & getName() const noexcept;
```